boards/arm/nrf5340dk_nrf5340/nrf5340_cpuapp_common.dts

```diff
@@ -101,6 +101,11 @@
 				 <5 &adc 5>;	/* A5 = P0.26 = AIN5 */
 	};
 
+	gpio_fwd: nrf-gpio-forwarder {
+		compatible = "nordic,nrf-gpio-forwarder";
+		status = "disabled";
```

Comment:
This is by default `"disabled"`. Will each module using forwarder which do not know about other modules using the forwarder need to set it to `"okay"`? Will it result in any conflict?

You provide in this PR update for thingy in file `boards/arm/thingy53_nrf5340/thingy53_nrf5340_common.dts` .
Can the two parts mentioned there `fem-gpio-if` and `fem-spi-if` come for example from two overlays which do not know about each other? What about status `"okay"`? Which one is responsible for setting it to `"okay"` ?

Reply:
You asked a couple of questions, so let me answer them one by one.
> Will each module using forwarder which do not know about other modules using the forwarder need to set it to `"okay"`?

Assuming that the board's definition itself doesn't set it to `"okay"` as e.g. Thingy53 does - yes, that's the safest approach.

> Will it result in any conflict?

As far as I can tell, no.

> Can the two parts mentioned there `fem-gpio-if` and `fem-spi-if` come for example from two overlays which do not know about each other?

Sure, actually I think it's going to be a common use case. Consider an nRF53 application that wants to use both a Front-End module and a Coexistence interface, and have the network core control them. The intended way to handle such a scenario using the solution from this PR would be for the Front-End module overlay to define its own child node of the `nrf-gpio-forwarder` and for the Coexistence interface overlay to define its own. These overlays might be some predefined shields, so in general they don't know about each other. When processing devicetree files for such an application, both these child nodes would end up in a single instance of `nrf-gpio-forwarder` and the initialization code will iterate over them both.

The only requirement for that to work is for both overlays to define child nodes with different names.

> What about status "okay"? Which one is responsible for setting it to "okay" ?

The way I imagined it is that every overlay that needs `nrf-gpio-forwarder` sets its status to `"okay"`.

Reply:
One more comment: I left the `nrf-gpio-forwarder` with status `"disabled"` by default because it seemed cleaner to me to have it disabled if it's unused. However, setting it to `"okay"` by default would simplify the overlays (the status wouldn't have to be modified at all) and decrease the chance of the application getting it wrong. Both solutions are fine IMO.

Reply:
I set the default status to `"okay"`, the solution seems easier to use this way.

Reply:
Usually for DT entries defined at SOC level status is set to `disabled` (it lists all possible UART, SPI, I2C instances). DT for board enables hardware used in given board (the UART instance connected to external module on the board).

Because `nrf-gpio-forwarder` is defined for board, I think it's OK to set it by default to `okay`.